### ISSUE TITLE
Command Line ...

### ISSUE THREAD

**Issue by tjm694** [elierodrigue/Hakchi_HelpingHand#1]
**KMFDManic commented 3 hours ago** 

BUG report...MISSING 2 vital command line arguments, that are crucial to running NES games compressed...

/bin/nes /CLV-H-CNXBG/CLV-H-CNXBG.nes.7z is MISSING

/bin/nes /usr/share/games/nes/kachikachi/CLV-H-CNXBG/CLV-H-CNXBG.nes.7z

as WELL as...at the end, --retroarch --fceumm OR --retroarch --core nestopia

like so...

/bin/nes /usr/share/games/nes/kachikachi/CLV-H-CNXBG/CLV-H-CNXBG.nes.7z --retroarch --fceumm

Games cannot run using retroarch in compressed form without these command line arguments.

You did an excellent job with the compression technique. So, I know it should be very easy to add these additional arguments as choices.. Choice between --retroarch --fceumm OR --retroarch --core nestopia, would also be very nice:)

MY post on reddit...

EDIT: The compression for the NEW Method WORKS great! But, it is missing a step (command line arguments!). I tested the compression! But, I did not realize it didn't take care of the command line arguments! So, here is how to fix it! I will inform him that he missed this vital step so that he can do an update to fix it! I apologize for any inconvenience!
Copy Exe into Games folder
Choose option to COMPRESS ALL GAMES
NES games require ONE more step...command line arguments that need added...Here is an example of 1942.
/bin/nes /CLV-H-THQIF/CLV-H-THQIF.nes.7z
AFTER /bin/nes/ you need to add /usr/share/games/nes/kachikachi, so it SHOULD be
/bin/nes /usr/share/games/nes/kachikachi/CLV-H-CNXBG/CLV-H-CNXBG.nes.7z
AND, at the VERY end, you need to choose the CORE, so it will be...
/bin/nes /usr/share/games/nes/kachikachi/CLV-H-CNXBG/CLV-H-CNXBG.nes.7z --retroarch --fceumm
or
/bin/nes /usr/share/games/nes/kachikachi/CLV-H-CNXBG/CLV-H-CNXBG.nes.7z --retroarch --core nestopia
You can ALSO do THIS for each game...
/bin/nes /usr/share/games/nes/kachikachi/CLV-H-CNXBG/CLV-H-CNXBG.nes.7z
MAKE sure to install RETROARCH HMOD, by itself! Then, you can install USE NESTOPIA, and that should take care of the END command line argument!
If anyone needs help fixing their set, private message me 

>  

> KMFDManic!

 I will help you out:) And, I am now going to post so he realizes he missed the command line arguments!

**Comment by KMFDManic** [comment]
Eh, thanks for copying my post to the right place, @tjm694

**Comment by elierodrigue** [comment]
WOW! I failed so hard! :P 
Testing a solution right now and actually validating on my nes mini for a change :)

**Comment by KMFDManic** [comment]
You didn't fail.  You actually did great on the coding.  I looked at how you coded it, and you did an exceptional job over the last month you've been working on it!  You even got it to retain the INFO for the games!  My compression method made reading the dat file for info impossible!  I posted a temporary fix to manually do the command line, and when you fix it, I will repost!  Please let me know what you changed in the code to do the adjustments:)  I am particularly interested in WHICH file/s you would need to adjust.  I would DEFINITELY recommend having --retroarch --fceumm  as the end argument, as fceumm is excellent for running these games.  BUT, if it isn't difficult to add a choice between --retroarch --fceumm OR --retroarch --core nestopia  THEN, awesome!  In any case, you did a fantastic job!  It did its job compressing.  Just need to add a few final touches to complete the process for working on the Final Flashed Mini NES:)

**Comment by elierodrigue** [comment]
In fact I coded everything yesterday :) I felt it was quite rush. Just tested my fix and its working. Now adding a core selection setting and i'll commit change + do a release.
I also added a live nes mini filebrowser / downloader.

**Comment by KMFDManic** [comment]
Awesome, I cant wait to try your master code out once done!

**Comment by KMFDManic** [comment]
I do have one other FEATURE Suggestion that you MAY be able to implement into your "Helping Hand".

I have a naming system I devised for EACH system, a PREFACE before each game name.  Examples are as follow...with SYSTEM abbreviation as a PREFACE, followed by a : (colon), then a space, THEN, the game's name:)

2600: Raiders of the Lost Ark
32X: Knuckles' Chaotix
7800: Commando
ARC: Shinobi
GB: Donkey Kong Land
GG: Tails Adventures
MD: Road Rash
NES: Tecmo Bowl
PCE: Darkwing Duck
PSX: Ridge Racer
SMS: Fantasy Zone
SNES: ActRaiser

Basically, once hakchi2 is opened, ALL systems are in nice, very organized order, and the games for each are alphabetical, with my PREFACE naming system.  Obviously, PREFACES can be used for GBC, GBA, NGP, and so on,  This PREFACE system has been completely awesome for my setup, and makes custom folders a breeze!  Especially with close to 2000 games!  Nothing is worse than having 2000 games in hakchi, all mixed up! Haha!

**Comment by elierodrigue** [comment]
Yeah, at some point I built a tabbed game browser for Hakchi but ClusterM didnt want to merge my pull request... I'll see if I can add the preface option. I might also add some folder management option in this software, something like : First level of folder is system. 
Then I also want to figure out a fix for the UI when there's not a lot of folders... Going left right is sometime really sketchy.

Well; I did the release, i'll close the issue. Feel free to reopen if you still have a problem with it!

**Comment by KMFDManic** [comment]
Fix for folder jumping, when not a lot of folders.  It seems, it would need at least 7-9 or more folders to not have the "jarring" effect.  You would maybe have to resize the space so that it is of lesser proportions and is unable to jump through the extra space.  Or, possibly make it so the folders repeat.  If only 3 folders, make them repeat in patterns of 3, then 3, then 3.  For 9 folders that may not jump.  There are a lot of ways to do it, creatively:)
